### PR TITLE
Hotkeys for target selection and run/walk

### PR DESCRIPTION
**Pull request by Cyberboss**
:cl: Cyberboss
add: NEW HOTKEYS
add: The number pad can be used to select the body part you want to target (Cycle through head -> eyes -> mouth with 8) in hotkey mode. Be sure Numlock is on!
add: Holding shift can be used as a modifier to your current run state in both hotkey and normal mode.
/:cl:

### CODE REVIEW

**Comment by ShadowDeath6** [comment]
run/walk could be something like 'C' since it's closer to WASD, and I don't think it's being used for anything else.

**Comment by CyberbossJHCB** [comment]
Can't use C and V cause it'd interfere with copy and paste

**Comment by Supermichael777** [comment]
Wait their are only 9 target zones why cant you make 7 and 9 eyes or mouth?

**Comment by CyberbossJHCB** [comment]
@Supermichael777 It's currently this way because of a coderbus vote. But I'll put up a strawpoll for a day and go with what that says.

**Comment by kevinz000** [comment]
What if I don't have a numpad? Also this basiclly nulls bulletproof armor if the attacker is experienced as they can just quickswitch targetting to hit limbs.

**Comment by PKPenguin321** [comment]
If you don't have a numpad you should get one

**Comment by MrPerson** [comment]
We don't need 9 fucking hotkeys for changing your limb target. We should be moving away from using it, not diving deeper down that hellhole.

**Comment by lzimann** [comment]
' ` ' is horrible to use in my keyboard =/

**Comment by CyberbossJHCB** [comment]
Closest alternatives for run/walk are b,h, and y

**Comment by ShadowDeath6** [comment]
>implying y isn't used
z users go home

**Comment by MrPerson** [comment]
The idea I've always had is you press shift to walk and you go back to running when you let go of it.

**Comment by CyberbossJHCB** [comment]
@MrPerson Difficult... but potentially doable

File: code/_onclick/hud/screen_objects.dm [hunk 1]
```diff
@@ -387,15 +391,22 @@
 				switch(icon_y)
```
Comment (Techhead0) [inline code/_onclick/hud/screen_objects.dm]: It's one line above this, but you missed the head.

**Comment by CyberbossJHCB** [comment]
Polls closed, current mode stays. Still investigating shift walking

**Comment by CyberbossJHCB** [comment]
I honestly don't think shift can work because

a) Because BYOND, we can't use shift directly, it's a modifier
b) The workaround means rewriting each moveproc for when shift is held and having it temporarily toggle move speed. But as a result, the icon state wouldn't toggle while just holding shift, only shift + movement.

So I'm open to doing one of three things.

1) Leave it on tilde.
2) Move it to 'B'
3) Removing it from the PR

Maintainers decide

**Comment by optimumtact** [comment]
b is just up from my spacebar where my hand usually sits, so I vote b

**Comment by CyberbossJHCB** [comment]
Oh god this diff, never using the online editor again

**Comment by optimumtact** [comment]
build failure was github dns related, restarted

**Comment by CyberbossJHCB** [comment]
Wait, @optimumtact, B is also resist

**Comment by optimumtact** [comment]
of course it is

**Comment by ShadowDeath6** [comment]
🅱

**Comment by MrPerson** [comment]
Shift will absolutely work, you just don't know how to do it. Please don't spread misinformation around

**Comment by optimumtact** [comment]
maybe if you had spent some time spreading the information there wouldn't be a vacuum for misinformation to fill

**Comment by CyberbossJHCB** [comment]
OK, I think we're done here

**Comment by nullbear** [comment]
Hot keys don't trigger if the screen is out of focus. Ctrl/paste is only used when the screen is out of focus.

**Pull request merged**